backend/api/communication/src/main/java/co/airy/core/api/communication/lucene/DocumentMapper.java

```diff
@@ -0,0 +1,114 @@
+package co.airy.core.api.communication.lucene;
+
+import co.airy.avro.communication.Channel;
+import co.airy.avro.communication.ChannelConnectionState;
+import co.airy.avro.communication.Message;
+import co.airy.core.api.communication.dto.Conversation;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.IntPoint;
+import org.apache.lucene.document.LongPoint;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+
+public class DocumentMapper {
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    public Document fromBytes(byte[] payload) {
+        final Conversation conversation;
+        try {
+            conversation = objectMapper.readValue(payload, Conversation.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return fromConversation(conversation);
+    }
+
+    public Document fromConversation(Conversation conversation) {
+        final Document document = new Document();
+        document.add(new StringField("id", conversation.getId(), Field.Store.YES));
+        document.add(new StringField("sourceConversationId", conversation.getSourceConversationId(), Field.Store.YES));
+
+        if (conversation.getDisplayName() != null) {
+            document.add(new TextField("display_name", conversation.getDisplayName(), Field.Store.YES));
+        }
+
+        document.add(new LongPoint("createdAt", conversation.getCreatedAt()));
+        document.add(new StoredField("createdAt", conversation.getCreatedAt()));
```

Comment:
Numeric points need to be stored separately as they are not returned in a query otherwise.